### PR TITLE
Use AppNotificationManager instead of ToastNotificationManager

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

 
Update Toast Windows to use the new API AppNotificationManager as Microsoft recommends: https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/adaptive-interactive-toasts?tabs=appsdk

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs [hunk 1]
```diff
@@ -20,7 +20,9 @@ public Snackbar()
 		VisualOptions = new SnackbarOptions();
 	}
 
-	/// <inheritdoc/>
+	/// <summary>
+	/// Returns true if the <see cref="Snackbar"/> is currently visible.
```
Comment (jfversluis) [inline src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs:24]: Details, but `true` could also be `<see langword="true"/>` to make it show up with a bit of syntax highlighting in IntelliSense, same for all keywords.

File: src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs [hunk 2]
```diff
@@ -49,14 +51,18 @@ public string ActionButtonText
 	/// <inheritdoc/>
 	public IView? Anchor { get; init; }
 
-	/// <inheritdoc/>
+	/// <summary>
+	/// Event occurred when <see cref="IsShown"/> changes.
```
Comment (jfversluis) [inline src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs:55]:
```suggestion
	/// Occurs when <see cref="IsShown"/> changes.
```
I would just go with "Occurs" people can already see that it's an event

File: src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs [hunk 3]
```diff
@@ -49,14 +51,18 @@ public string ActionButtonText
...
+	/// </summary>
 	public static event EventHandler Shown
 	{
 		add => weakEventManager.AddEventHandler(value);
 		remove => weakEventManager.RemoveEventHandler(value);
 	}
 
-	/// <inheritdoc/>
+	/// <summary>
+	/// Event occurred when Snackbar dismisses.
```
Comment (jfversluis) [inline src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs:64]:
```suggestion
	/// Occurs when Snackbar is dismissed.
```

**Review by jfversluis — APPROVED** [review]
Couple of very minor things! The actual code looks good to me!

File: src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs (hunk 1 above)
Comment (VladislavAntonyuk) [reply src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs:24]:
```suggestion
	/// Returns <see langword="true"/> if the <see cref="Snackbar"/> is currently visible.
```

File: src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs [hunk 4]
```diff
@@ -72,20 +72,36 @@ public double TextSize
...
+#endif
+	
 	/// <summary>
 	/// Dismiss Toast
 	/// </summary>
+#if WINDOWS
+	public virtual async Task Dismiss(CancellationToken token = default)
+	{
+		await DismissPlatform(token);
+		await Task.CompletedTask;
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs:96]: why are we waiting here?
```suggestion
```

File: src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs [hunk 5]
```diff
@@ -72,20 +72,36 @@ public double TextSize
 	/// <summary>
 	/// Show Toast
 	/// </summary>
+#if WINDOWS
+	public virtual async Task Show(CancellationToken token = default)
+	{
+		await ShowPlatform(token);
+		await Task.CompletedTask;
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Alerts/Toast/Toast.shared.cs:79]: why are we waiting here?
```suggestion
```

File: src/CommunityToolkit.Maui/Alerts/Toast/Toast.windows.cs [hunk 6]
```diff
@@ -24,28 +26,34 @@ protected virtual void Dispose(bool isDisposing)
 		isDisposed = true;
 	}
 
-	static void DismissPlatform(CancellationToken token)
+	static async Task DismissPlatform(CancellationToken token)
 	{
 		if (PlatformToast is not null)
```
Comment (pictos) [inline src/CommunityToolkit.Maui/Alerts/Toast/Toast.windows.cs:31]: can we invert this `if` for better readability?

**Review by pictos — APPROVED** [review]

File: src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs [hunk 7]
```diff
@@ -8,8 +8,8 @@ public partial class Snackbar : ISnackbar
 	static readonly WeakEventManager weakEventManager = new();
 
 	bool isDisposed;
-	string text = string.Empty;
-	string actionButtonText = AlertDefaults.ActionButtonText;
+	readonly string text = string.Empty;
+	readonly string actionButtonText = AlertDefaults.ActionButtonText;
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.shared.cs:12]: Cool!! I didn't know that we could use `readonly` fields in `get; init;` properties! 

I like it. It makes a lot of sense 💯

File: src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs [hunk 8]
```diff
@@ -8,19 +8,6 @@ namespace CommunityToolkit.Maui.Extensions;
 /// </summary>
 public static class ToastNotificationExtensions
 {
-	/// <summary>
-	/// Build Toast Text Notification
-	/// </summary>
-	/// <param name="text">Notification text</param>
-	/// <returns><see cref="XmlDocument"/></returns>
-	public static XmlDocument BuildToastNotificationContent(string text)
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs:16]: FYI - this is a breaking change, removing a public API

We'll need to bump our Major version number for our next release (which'll hopefully coincide with the release of `MediaPlayer`)

Comment (brminnick) [reply src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs:16]: That being said, I'd be surprised if removing this API impacts any devs

**Comment by HASSA17Y** [comment]
[]()@

File: src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs (hunk 8 above)
Comment (HASSA17Y) [reply src/CommunityToolkit.Maui/Extensions/ToastNotificationExtensions.windows.cs:16]: How start coding

**Pull request merged**